bpfd/src/command.rs

```diff
@@ -89,6 +89,41 @@ pub(crate) enum Location {
     File(String),
 }
 
+impl Location {
```

Comment:
Looking at this more... I __think__ it'd be best to only have one function here called `get_program_bytes` it can do all of the verification in one step that we're currently doing in a few steps and a few different places and do it down in `add_*_program` i.e something along the lines of this sudo code (not checked for bugs :))

``` rust
fn get_program_bytes(&self, provided_section_name: String) -> Result<Vec<u8>, BpfdError> {
    match self {
        Location::File(l) => read(l).await,
        Location::Image(l) => {
            let program_overrides = l
                .get_image(None)
                .await
                .map_err(|e| BpfdError::BpfBytecodeError(e.into()))?;

            // If section name isn't provided and we're loading from a container
            // image use the section name provided in the image metadata, otherwise
            // always use the provided section name.
            if section_name.is_empty() {
                section_name = program_overrides.image_meta.section_name
            } else if program_overrides.image_meta.section_name != section_name {
                return Err(BpfdError::BytecodeMetaDataMismatch {
                    image_sec_name: program_overrides.image_meta.section_name,
                    provided_sec_name: section_name,
                });
            }
            get_bytecode_from_image_store(program_overrides.path).await
        }
}
```

That can be called [here](https://github.com/bpfd-dev/bpfd/blob/main/bpfd/src/bpf.rs#L160) in bpf.rs and tc.rs/xdp.rs which should condense all of this logic

Reply:
Thanks for comments! Trying to implement (pushed).